src/shapes/Object/Object.ts

```diff
@@ -1360,7 +1360,15 @@ export class FabricObject<
    * @param {Boolean} [options.viewportTransform] Account for canvas viewport transform
    * @return {HTMLCanvasElement} Returns DOM element <canvas> with the FabricObject
    */
-  toCanvasElement(options: any = {}) {
+  toCanvasElement(
+    options: any = {},
+    createToCanvas = (el: HTMLCanvasElement) =>
```

Comment:
can we rename the arg?
maybe to `createCanvas`? I don't know.
I dislike the way this method is impl.
A lot of mess, especially the fact that we are mutating the ref itself instead of cloning and returning a promise.
Maybe that is a better impl.

1. clone
2. send to canvas plane so that tl is positioned at 0,0
3. create canvas
4. render
5. return both canvas and clone

Thoughts?
For now this is a good enough patch

Reply:
Also the setCoords calls seem to me quite redundant unless the object uses them for the exported image

Reply:
and if that is the case I would argue the dev should subclass and call setCoords themselves

Reply:
setCords is there to avoid offScreen check issues.
i **THINK** it can be removed by specifying for the new canvas that it should not use the offscreen check.
I do not think there is any other reason

But let's do it in a different PR